app/models/manageiq/providers/redhat/inventory/persister/target_collection.rb

```diff
@@ -0,0 +1,75 @@
+class ManageIQ::Providers::Redhat::Inventory::Persister::TargetCollection < ManageIQ::Providers::Redhat::Inventory::Persister
+  def initialize_inventory_collections
+    add_targeted_inventory_collections
+    add_remaining_inventory_collections([infra], :strategy => :local_db_find_references)
+
+    # TODO: check what needs to be added here
+  end
+
+  def references(collection)
+    target.manager_refs_by_association.try(:[], collection).try(:[], :ems_ref).try(:to_a) || []
+  end
+
+  # TODO: check whether needed
+  def name_references(collection)
+    target.manager_refs_by_association.try(:[], collection).try(:[], :name).try(:to_a) || []
```

Comment:
I need to refactor these, it's basically just about extracting IDs of some kinds from all the targets we have, the current structure is not nice